src/Credentials/InstanceProfileProvider.php

```diff
@@ -52,8 +62,22 @@ public function __invoke()
             if (!$this->profile) {
                 $this->profile = (yield $this->request(self::CRED_PATH));
             }
-            $json = (yield $this->request(self::CRED_PATH . $this->profile));
-            $result = $this->decodeResult($json);
+            $result = null;
+            while ($result == null) {
+                try {
+                    $json = (yield $this->request(self::CRED_PATH . $this->profile));
+                    $result = $this->decodeResult($json);
+                } catch (InvalidJsonException $e) {
```

Comment:
Should there be a retry for things beyond invalid JSON?

Reply:
I think for now, we should limit this to the JSON error.  The original issue that precipitated needing to retry was for invalid JSON.  We may want to extend this to throttling errors in the future, but will want to be careful to not exacerbate that issue.